ncm-openstack/src/main/pan/components/openstack/common.pan

```diff
@@ -0,0 +1,155 @@
+# ${license-info}
+# ${developer-info}
+# ${author-info}
+
+
+declaration template components/openstack/common;
+
+type type_storagebackend = string with match(SELF, '^(file|http|swift|rbd|sheepdog|cinder|vmware)$');
+
+type type_neutrondriver = string with match(SELF, '^(local|flat|vlan|gre|vxlan|geneve)$');
+
+type type_neutronextension = string with match(SELF, '^(qos|port_security)$');
+
+type type_directory = string with match(SELF, '[^/]+/?$');
+
+@documentation {
+    OpenStack common domains section
+}
+type openstack_domains_common = {
+    @{Domain name containing project}
+    'project_domain_name' : string = 'Default'
+    @{Project name to scope to}
+    'project_name' : string = 'service'
```

Comment:
that is not a good default. i think it is a very specific project to manage the services, not to be used by anything else